internal/subscriber/model.go

```diff
@@ -151,25 +172,63 @@ func (a *AckMessage) HasHitDeadline() bool {
 }
 
 // ParseAckID ...
-func ParseAckID(ackID string) *AckMessage {
+func ParseAckID(ackID string) (*AckMessage, error) {
 	// split the message and parse tokens
 	parts := strings.Split(ackID, ackIDSeparator)
+	decodedParts := decode(parts)
 
-	// TODO : add validations
-	partition, _ := strconv.ParseInt(utils.Decode(parts[3]), 10, 0)
-	offset, _ := strconv.ParseInt(utils.Decode(parts[4]), 10, 0)
-	deadline, _ := strconv.ParseInt(utils.Decode(parts[5]), 10, 0)
+	if !isValidAckID(decodedParts) {
+		return nil, errInvalidAckID
+	}
+
+	partition, err := strconv.ParseInt(decodedParts[3], 10, 0)
+	if err != nil {
+		err = errors.WithMessage(err, fmt.Sprintf("error in parsing partition value %v to int", decodedParts[3]))
```

Comment:
```suggestion
		return nil, errors.WithMessage(err, fmt.Sprintf("error in parsing partition value %v to int", decodedParts[3]))
```
nit: could have done away with the redundant line to init the error 👀 what say?

Reply:
I think in one of the previous comments, you suggested to take out the error and initialise it in a var and use that

[https://github.com/razorpay/metro/pull/242#discussion_r703400310](url)

Reply:
That was when you were logging the same error as well. Then init once and re-using the `err var` in the `logline` and in the `return` made sense. Now that we have removed the logline we can simply return.

Reply:
My previous comment got deleted somehow. Re-posting again.

Previously we were logging and returning the same err hence keeping in a var made sense to avoid duplicates. Now we have removed the logger so a direct return makes sense.